pkg/virtctl/adm/logverbosity/logverbosity_test.go

```diff
@@ -0,0 +1,279 @@
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/types"
+
+	"kubevirt.io/kubevirt/tests/clientcmd"
+
+	"kubevirt.io/client-go/kubecli"
+
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+)
+
+const (
+	NAMESPACE = "kubevirt"
+	NAME      = "kubevirt"
+)
+
+type virtComponent int
+
+const (
+	virtAPI virtComponent = iota // virtAPI must be at the first position because it is used for the iteration
+	virtController
+	virtHandler
+	virtLauncher
+	virtOperator
+	all // all must be at the end, because it is used for the iteration
+)
+
+const virtComponentNum = int(all) + 1 // number of virt components
+
+var _ = Describe("Log Verbosity", func() {
+
+	var ctrl *gomock.Controller
+	var kvInterface *kubecli.MockKubeVirtInterface
+
+	var kv *v1.KubeVirt
+	var kvs *v1.KubeVirtList
+
+	BeforeEach(func() {
+
+		// create mock KubeVirt CR
+		kv = NewKubeVirtWithoutVerbosity(NAMESPACE, NAME)
+		kvs = kubecli.NewKubeVirtList(*kv)
+
+		// create the wrapper that would return the mock virt client to the code being unit tested
+		ctrl = gomock.NewController(GinkgoT())
+		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+
+		// create mock interface (clientset)
+		kvInterface = kubecli.NewMockKubeVirtInterface(ctrl)
+
+		// set up mock client bahavior
+		kubecli.MockKubevirtClientInstance.EXPECT().KubeVirt(NAMESPACE).Return(kvInterface).AnyTimes()
+		kubecli.MockKubevirtClientInstance.EXPECT().KubeVirt("").Return(kvInterface).AnyTimes()
+
+		// set up mock interface behavior
+		kvInterface.EXPECT().Get(NAME, gomock.Any()).Return(kv, nil).AnyTimes()
+		kvInterface.EXPECT().List(gomock.Any()).Return(kvs, nil).AnyTimes()
+		kvInterface.EXPECT().Patch(NAME, types.JSONPatchType, gomock.Any(), gomock.Any()).DoAndReturn(
+			func(_ any, _ any, patchData []byte, _ any, _ ...any) (*v1.KubeVirt, error) {
+				patch, err := jsonpatch.DecodePatch(patchData)
+				Expect(err).ToNot(HaveOccurred())
+				kvJSON, err := json.Marshal(kv)
+				Expect(err).ToNot(HaveOccurred())
+				modifiedKvJSON, err := patch.Apply(kvJSON)
+				Expect(err).ToNot(HaveOccurred())
+
+				// reset the object in preparation for unmarshal,
+				// since unmarshal does not guarantee that fields in kv will be removed by the patch
+				kv = &v1.KubeVirt{}
+
+				err = json.Unmarshal(modifiedKvJSON, kv)
+				Expect(err).ToNot(HaveOccurred())
+				return kv, nil
+			}).AnyTimes()
+	})
+
+	When("with invalid set of flags", func() {
+		Context("with empty set of flags", func() {
+			It("should fail (return help)", func() {
+				cmd := clientcmd.NewRepeatableVirtctlCommand("adm", "log-verbosity")
+				err := cmd()
+				Expect(err).NotTo(Succeed())
+				Expect(err).To(MatchError(ContainSubstring("no flag specified - expecting at least one flag")))
+			})
+		})
+
+		DescribeTable("should fail handled by cobra", func(args ...string) {
```

Comment:
nit: please, don't put `cobra` in the description, this is just the package we use for creating the cli

Reply:
I'll change it to "should fail handled by the CLI package".

Reply:
Done. Please confirm.